src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/GeneralizedCoastlineCheck.java

```diff
@@ -15,27 +17,35 @@
 import org.openstreetmap.atlas.tags.NaturalTag;
 import org.openstreetmap.atlas.tags.annotations.validation.Validators;
 import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Angle;
 import org.openstreetmap.atlas.utilities.scalars.Distance;
+import org.openstreetmap.atlas.utilities.tuples.Tuple;
 
 /**
  * Flags generalized coastlines-- that is, OSM ways with the tag natural=coastline where x% or more
  * node pairs are y or more meters apart. X and y can be specified in a {@link Configuration}.
- * Coastlines can be represented in Atlas by {@link LineItem}s and may be members of
- * {@link Relation}s.
+ * Additionally flags sharp angle locations according to the angle.minimum.threshold
+ * {@link Configuration} value if it's greater than -1. Coastlines can be represented in Atlas by
+ * {@link LineItem}s and may be members of {@link Relation}s.
  *
  * @author seancoulter
  */
 public class GeneralizedCoastlineCheck extends BaseCheck<Long>
 {
-    private static final List<String> FALLBACK_INSTRUCTIONS = Collections.singletonList(
-            "This coastline is generalized, as {0}% of node pairs are {1} or more meters apart. To fix, increase the number of nodes along this coastline. The midpoint of generalized segments is dotted for convenience.");
+    private static final String BASIC_INSTRUCTIONS = "This coastline is generalized, as {0}% of node pairs are {1} or more apart. To fix, add more nodes to this coastline. The midpoints of generalized segments are dotted for convenience.";
+    private static final String SHARP_ANGLE_INSTRUCTIONS = "This coastline is generalized, as {0}% of node pairs are {1} or more apart. There are also sharp angles exceeding {2} degrees. To fix, add more nodes to smooth angles and break up long segments of coastline. Suggested areas to add nodes are dotted.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(BASIC_INSTRUCTIONS,
+            SHARP_ANGLE_INSTRUCTIONS);
     private static final double MINIMUM_DISTANCE_BETWEEN_NODES = 100;
     private static final double MINIMUM_NODE_PAIR_THRESHOLD_PERCENTAGE = 30.0;
+    private static final double SHARP_ANGLE_THRESHOLD_DEFAULT = Integer.MAX_VALUE;
```

Comment:
Why Max value here?

Reply:
It's supposed to be a value that shouldn't ever be used in the configuration. It could technically be anything over 359 degrees but I think this makes it clearest